File: app/src/main/java/team/triplog/presentation/fragment/MainTripPlanFragment.java
package team.triplog.presentation.fragment;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.constraintlayout.widget.Group;
import androidx.core.app.ActivityOptionsCompat;
import androidx.core.util.Pair;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Objects;

import team.triplog.R;
import team.triplog.presentation.activity.TripPlanActivity;
import team.triplog.presentation.adapter.MainTripPlanAdapter;
import team.triplog.domain.entity.Trip;

@SuppressLint("SimpleDateFormat")
public class MainTripPlanFragment extends Fragment implements MainTripPlanAdapter.OnItemClickListener {
    private SimpleDateFormat periodFormat = new SimpleDateFormat("yyyy.MM.dd");
    private View rootView;
    private Group groupNoContents;
    private RecyclerView recyclerTripPlan;
    private MainTripPlanAdapter mainTripPlanAdapter;
    private ArrayList<Trip> trips = new ArrayList<>();

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        rootView = inflater.inflate(R.layout.fragment_main_trip_plan, container, false);

        init();
        setUi();

        return rootView;
    }

    private void init() {
        groupNoContents = rootView.findViewById(R.id.group_no_contents);
        recyclerTripPlan = rootView.findViewById(R.id.recycler_trip_plan);

        // TODO : Test code
        trips.add(new Trip());
        trips.add(new Trip());
        trips.add(new Trip());

        mainTripPlanAdapter = new MainTripPlanAdapter(trips, this);
        recyclerTripPlan.setLayoutManager(new LinearLayoutManager(getContext()));
        recyclerTripPlan.setAdapter(mainTripPlanAdapter);
    }

    private void setUi() {
        if (isAdded()) {
            if (trips.isEmpty()) {
                groupNoContents.setVisibility(View.VISIBLE);
            } else {
                groupNoContents.setVisibility(View.INVISIBLE);
            }
        }
    }

    @Override
    public void onItemClick(View view, int position) {
        Trip trip = trips.get(position);

        Intent intent = new Intent(getContext(), TripPlanActivity.class);

        // TODO : 임시로 String 형식으로 넘김.
        intent.putExtra("EXTRA_TRIP_NAME", trip.name);
        intent.putExtra("EXTRA_TRIP_PERIOD", String.format(
                "%s - %s",
                periodFormat.format(trip.startDate),
                periodFormat.format(trip.startDate)
        ));

        View layoutTrip = view.findViewById(R.id.layout_item);

        Pair<View, String> pairItem = Pair.create(layoutTrip, layoutTrip.getTransitionName());

        ActivityOptionsCompat optionsCompat = ActivityOptionsCompat.makeSceneTransitionAnimation(
                Objects.requireNonNull(getActivity()),
                pairItem
        );

        startActivity(intent, optionsCompat.toBundle());
    }
}

File: app/src/main/java/team/triplog/presentation/fragment/MainTripPlanFragment.kt
package team.triplog.presentation.fragment

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.app.ActivityOptionsCompat
import androidx.core.util.Pair
import androidx.fragment.app.Fragment
import team.triplog.R
import team.triplog.databinding.FragmentMainTripPlanBinding
import team.triplog.domain.model.Trip
import team.triplog.presentation.trip.plan.activity.TripPlanActivity
import team.triplog.presentation.adapter.MainTripPlanAdapter
import java.text.SimpleDateFormat
import java.util.*

class MainTripPlanFragment : Fragment(), MainTripPlanAdapter.OnItemClickListener {
    private lateinit var binding: FragmentMainTripPlanBinding

    private val periodFormat = SimpleDateFormat("yyyy.MM.dd", Locale.getDefault())

    private var mainTripPlanAdapter: MainTripPlanAdapter? = null
    private val trips = ArrayList<Trip>()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        binding = FragmentMainTripPlanBinding.inflate(inflater, container, false)
        binding.lifecycleOwner = this

        init()
        setUi()

        return binding.root
    }

    private fun init() {
        // TODO : Test code
        trips.add(Trip())
        trips.add(Trip())
        trips.add(Trip())

        mainTripPlanAdapter = MainTripPlanAdapter(trips, this)
        binding.recyclerTripPlan.adapter = mainTripPlanAdapter
    }

    private fun setUi() {
        if (isAdded) {
            binding.groupNoContents.visibility = if (trips.isEmpty()) View.VISIBLE else View.INVISIBLE
        }
    }

    override fun onItemClick(view: View, item: Trip) {
        val intent = Intent(context, TripPlanActivity::class.java)

        // TODO : 임시로 String 형식으로 넘김.
        intent.putExtra("EXTRA_TRIP_NAME", item.name)
        intent.putExtra(
            "EXTRA_TRIP_PERIOD", String.format(
                "%s - %s",
                periodFormat.format(item.startDate),
                periodFormat.format(item.startDate)
            )
        )
        val viewTrip = view.findViewById<View>(R.id.view_trip)
        val pairItem = Pair.create(viewTrip, viewTrip.transitionName)
        val optionsCompat =
            ActivityOptionsCompat.makeSceneTransitionAnimation(Objects.requireNonNull(requireActivity()), pairItem)

        startActivity(intent, optionsCompat.toBundle())
    }
}